### PR TITLE
Post Methods - Try/Except Errors

### PR DESCRIPTION
**Pull request by evymai**
Addresses the server-side functionality of NSS-Day-Cohort-68/rare-client-rare-team-1

The purpose of this PR is to update the POST methods to:

- catch user input errors
- address incorrect naming conventions
- improve HTTP error responses 

**Changes made in json-server.py**

- added try/except block before executing create_category, create_comment, create_post, create_tag, and create_posttag to check if the required keys are provided by the client. If not the KeyError is caught and an error message is returned to the client.

**Steps to test**

- For each category - attempt to to post an object with:
-  expected keys, extra keys: should be successful
- missing values, no body: should have an error response

### CODE REVIEW

**Review by AKnepshield — APPROVED** [review]
Allows user to successfully create a category and throws a error if input fields are incorrect.  Works as described.

File: json-server.py [hunk 1]
```diff
@@ -62,59 +62,110 @@ def do_POST(self):
             )
 
         if url["requested_resource"] == "comments":
+            try:
```
Comment (clairecatohanson) [inline json-server.py:65]: tested and works as designed, looks great!

File: json-server.py [hunk 2]
```diff
@@ -62,59 +62,110 @@ def do_POST(self):
...
                     "Successfully created", status.HTTP_201_SUCCESS_CREATED.value
                 )
-
             return self.response(
-                "Invalid data", status.HTTP_400_CLIENT_ERROR_BAD_REQUEST_DATA.value
+                "An unexpected error occurred.", status.HTTP_500_SERVER_ERROR.value
             )
 
         if url["requested_resource"] == "categories":
+            try:
```
Comment (clairecatohanson) [inline json-server.py:85]: tested and works as designed, looks great!

File: json-server.py [hunk 3]
```diff
@@ -62,59 +62,110 @@ def do_POST(self):
...
                     status.HTTP_201_SUCCESS_CREATED.value,
                 )
-
             return self.response(
-                "Invalid data", status.HTTP_400_CLIENT_ERROR_BAD_REQUEST_DATA.value
+                "An unexpected error occurred.", status.HTTP_500_SERVER_ERROR.value
             )
 
         if url["requested_resource"] == "tags":
+            try:
```
Comment (clairecatohanson) [inline json-server.py:106]: tested and works as designed, looks great!

File: json-server.py [hunk 4]
```diff
@@ -62,59 +62,110 @@ def do_POST(self):
...
                     "Successfully created", status.HTTP_201_SUCCESS_CREATED.value
                 )
-
             return self.response(
-                "Invalid data", status.HTTP_400_CLIENT_ERROR_BAD_REQUEST_DATA.value
+                "An unexpected error occurred.", status.HTTP_500_SERVER_ERROR.value
             )
 
         if url["requested_resource"] == "posts":
+            try:
```
Comment (clairecatohanson) [inline json-server.py:126]: tested and works as designed, looks great!

File: json-server.py [hunk 5]
```diff
@@ -62,59 +62,110 @@ def do_POST(self):
...
                     "Successfully created", status.HTTP_201_SUCCESS_CREATED.value
                 )
-
             return self.response(
-                "Invalid data", status.HTTP_400_CLIENT_ERROR_BAD_REQUEST_DATA.value
+                "An unexpected error occurred.", status.HTTP_500_SERVER_ERROR.value
             )
 
         if url["requested_resource"] == "posttags":
+            try:
```
Comment (clairecatohanson) [inline json-server.py:152]: tested and works as designed, looks great!

**Review by clairecatohanson — APPROVED** [review]
Tested comment, category, tag, post, and posttags. All try/except blocks work as expected and follow naming conventions.

**Pull request merged**